torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -1041,12 +1041,38 @@ class CppMicroGemmAMX(CppMicroGemm):
         for (int idx_dq = 0, idx_q = 0; idx_dq < buf_size; idx_q += ldb, idx_dq += {{block_n}}) {
         {%- for vec_idx in range(0, block_n, 32) %}
             {%- if (block_n - vec_idx) >= 32 %}
-            auto b_int8_idx_{{vec_idx}} = at::vec::Vectorized<int8_t>::loadu(
-                base_addr + idx_q + {{vec_idx}} ,
-                static_cast<int64_t>(32)
-            );
-            auto b_bf16_idx_{{vec_idx}} = at::vec::convert<{{input_t}}>(b_int8_idx_{{vec_idx}});
-            b_bf16_idx_{{vec_idx}}.store(dequantized_B_buf + idx_dq + {{vec_idx}});
+            // 1) Load 32 x int8
+            __m256i v8  = _mm256_loadu_si256((const __m256i*)(base_addr + idx_q + {{vec_idx}}));
+            // 2) Widen: 32 x i8 -> 32 x i16
+            __m512i v16 = _mm512_cvtepi8_epi16(v8);  // sign-extend. Use _mm512_cvtepu8_epi16 for unsigned
+            // Split the 32 x i16 into two 16-lane halves
+            __m256i v16_lo = _mm512_castsi512_si256(v16);
+            __m256i v16_hi = _mm512_extracti64x4_epi64(v16, 1);
+            // 3) Widen each half to i32
+            __m512i v32_lo = _mm512_cvtepi16_epi32(v16_lo);
+            __m512i v32_hi = _mm512_cvtepi16_epi32(v16_hi);
+            // 4) Convert to f32
+            __m512 f_lo = _mm512_cvtepi32_ps(v32_lo);
+            __m512 f_hi = _mm512_cvtepi32_ps(v32_hi);
+            // 5) f32 -> bf16 (round-to-nearest-even) and pack 32 lanes to 512b
+            // Packs the second operand (f_lo) into the lower 16 bf16 lanes and the first (f_hi) into the upper 16.
+            __m512i bf = (__m512i)_mm512_cvtne2ps_pbh(f_hi, f_lo);
+            // 6) Store 32 x bf16 (512 bits)
+            _mm512_storeu_si512((__m512i*)(dequantized_B_buf + idx_dq + {{vec_idx}}), bf);
+            {%- elif (block_n - vec_idx) >= 16 %}
+            // 1) Load 16 x int8 (128 bits)
+            __m128i v8 = _mm_loadu_si128((const __m128i*)(base_addr + idx_q + {{vec_idx}}));
+            // 2) Widen: 16 x i8 -> 16 x i16
+            __m256i v16 = _mm256_cvtepi8_epi16(v8);   // for signed
+            // use _mm256_cvtepu8_epi16 for unsigned
+            // 3) Widen further: 16 x i16 -> 16 x i32
+            __m512i v32 = _mm512_cvtepi16_epi32(v16);
+            // 4) Convert to f32
+            __m512 f32 = _mm512_cvtepi32_ps(v32);
+            // 5) Convert f32 -> bf16 (round-to-nearest-even)
+            __m256i bf16 = (__m256i)_mm512_cvtneps_pbh(f32);
```

Comment:
Since intrinsic is used, It is better to check whether the compiler supports these, e.g., `_mm512_cvtneps_pbh`.
If the compiler does not support, it will choose aten linear.
Maybe we can do like https://github.com/pytorch/pytorch/pull/147368.